pkg/clusteragent/admission/mutate/auto_instrumentation.go

```diff
@@ -371,25 +411,25 @@ func injectAutoInstruConfig(pod *corev1.Pod, libsToInject []libInfo) error {
 		err := injectLibInitContainer(pod, image, lang)
 		if err != nil {
 			langStr := string(lang)
-			metrics.LibInjectionErrors.Inc(langStr)
-			metrics.MutationErrors.Inc(metrics.LibInjectionMutationType, "cannot inject init container", langStr)
+			metrics.LibInjectionErrors.Inc(langStr, strconv.FormatBool(autoDetected))
+			metrics.MutationErrors.Inc(metrics.LibInjectionMutationType, "cannot inject init container", langStr, strconv.FormatBool(autoDetected))
 			lastError = err
 			log.Errorf("Cannot inject init container into pod %s: %s", podString(pod), err)
 		}
 		err = injectLibConfig(pod, lang)
 		if err != nil {
 			langStr := string(lang)
-			metrics.LibInjectionErrors.Inc(langStr)
-			metrics.MutationErrors.Inc(metrics.LibInjectionMutationType, "cannot inject lib config", langStr)
+			metrics.LibInjectionErrors.Inc(langStr, strconv.FormatBool(autoDetected))
+			metrics.MutationErrors.Inc(metrics.LibInjectionMutationType, "cannot inject lib config", langStr, strconv.FormatBool(autoDetected))
 			lastError = err
 			log.Errorf("Cannot inject library configuration into pod %s: %s", podString(pod), err)
 		}
 	}
 
 	// try to inject all if the annotation is set
 	if err := injectLibConfig(pod, "all"); err != nil {
```

Comment:
should we only do this if there weren't any languages?

Reply:
This will happen in case `"admission.datadoghq.com/all-lib.config.v1"` is set on the pod. (check [here](https://github.com/DataDog/datadog-agent/blob/749288b6b8487141073e82657c681629a94f068e/pkg/clusteragent/admission/mutate/auto_instrumentation.go#L542-L545)).

In some situations, it is intended that the user wants to inject all supported libraries instead of specifying a selection of languages in the annotations. This PR helps improve this in the future by allowing the agent to auto-detect languages used in the user application, and injecting the libraries of the detected languages only. For now, the feature is still in beta, and the default behaviour remains unchanged for now.